### PR TITLE
Do not select disabled items

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Fixes an issue noticed in https://github.com/adobe/react-spectrum/pull/3123 where dragging a disabled item to a new list where it was still disabled caused it to become selected.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/list/test/ListView.test.js [hunk 1]
```diff
@@ -1231,523 +1223,4 @@ describe('ListView', function () {
       expect(grid.scrollTop).toBe(0);
     });
   });
-
-  describe('drag and drop', function () {
```
Comment (snowystinger) [inline packages/@react-spectrum/list/test/ListView.test.js:1235]: Removed drag and drop into its own file so this file wouldn't end up like TableView's tests

File: packages/@react-spectrum/list/test/ListViewDnD.test.tsx [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+          type: 'dragend',
+          keys: new Set(['a', 'b', 'c', 'd']),
+          x: 50,
+          y: 25,
+          dropOperation: 'move'
+        });
+      });
+
+
+      it('should not select disabled items', async function () {
```
Comment (snowystinger) [inline packages/@react-spectrum/list/test/ListViewDnD.test.tsx:444]: This is the only new test I added

**Review by LFDanLu — APPROVED** [review]
LGTM, thanks for organizing the tests!

**Comment by snowystinger** [comment]
Closing as https://github.com/adobe/react-spectrum/pull/3123 also handled it

**Pull request closed without merging**